File: intellij-lsp/src/com/github/gtache/lsp/settings/gui/LSPGUI.scala
package com.github.gtache.lsp.settings.gui

import javax.swing.JPanel

import com.github.gtache.lsp.settings.LSPState

trait LSPGUI {

  import LSPGUI.lspState

  def state: LSPState = lspState

  def isModified: Boolean

  def reset(): Unit

  def apply(): Unit

  def getRootPanel: JPanel

}

object LSPGUI {
  val lspState: LSPState = LSPState.getInstance()
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/settings/gui/LSPGUI.kt
package com.github.gtache.lsp.settings.gui

import javax.swing.JPanel

/**
 * Represents a settings GUI
 */
interface LSPGUI {

    /**
     * Whether the values are modified or not
     */
    fun isModified(): Boolean

    /**
     * Resets to the initial values
     */
    fun reset(): Unit

    /**
     * Applies the selected values
     */
    fun apply(): Unit

    /**
     * Returns the root panel
     */
    fun getRootPanel(): JPanel

}